.github/workflows/sonarcloud.yml

```
@@ -0,0 +1,102 @@
+        run: |
+          status=0
+          for dir in project/backend/*/; do
+            if [ -f "$dir/build.gradle" ] && [ -f "$dir/gradlew" ]; then
+              echo "Running Sonar analysis for $dir"
+              (cd "$dir" && ./gradlew test jacocoTestReport sonarqube --info) || {
+                echo "WARNING: Analysis for $dir failed but continuing with other services"
+                status=1
+              }
+            fi
+          done
+          exit $status
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)